service/storage_service.cc

```diff
@@ -7193,7 +7193,20 @@ future<join_node_response_result> storage_service::join_node_response_handler(jo
             slogger.info("raft topology: coordinator accepted request to join, "
                     "waiting for nodes {} to be alive before responding and continuing",
                     sync_nodes);
-            co_await _gossiper.wait_alive(sync_nodes, wait_for_live_nodes_timeout);
+            try {
+                co_await _gossiper.wait_alive(sync_nodes, wait_for_live_nodes_timeout);
+            } catch (std::runtime_error& e) {
+                const sstring message = ::format(
+                        "waiting for nodes {} to be alive failed after coordinator accepted request to join: {}. "
+                        "Make sure all nodes are alive before restarting this node.", sync_nodes, e.what());
+                slogger.warn("raft topology: {}", message);
+                e = std::runtime_error(message);
```

Comment:
Since the exception handler should cover the entirety of the `overloaded_functor` in the `join_node_response_handler`, I will just rethrow the caught exception, which will be of any type. Probably, only `abort_requested_exception` will require an own `catch` statement. So, this comment and the comment below will not be relevant anymore.

Thank you for the detailed review anyway. It helped me understand a few things.